viz_scripts/plots.py

```diff
@@ -264,18 +264,25 @@ def energy_impact(x,y,color,plot_title,file_name):
     plt.legend(labels=objects, handles=patches, loc='upper right', borderaxespad=0, fontsize=15, frameon=True)
     plt.savefig(SAVE_DIR+file_name+".png", bbox_inches='tight')
 
-def barplot_mode(data,x,y,plot_title, labels, file_name):
+def barplot_mode(data,x,y,x_label,plot_title, labels, file_name, values_to_translations={}):
     colours = dict(zip(labels, plt.cm.tab20.colors[:len(labels)]))
     sns.set(font_scale=1.5)
     f = plt.subplots(figsize=(15, 6))
     sns.set(style='whitegrid')
     ax = sns.barplot(x=x, y=y, palette=colours,data=data, ci=None)
-    plt.xlabel(x, fontsize=23)
+    plt.xlabel(x_label, fontsize=23)
     plt.ylabel(y, fontsize=23)
     plt.title(plot_title, fontsize=25)
+
+    if values_to_translations:
+        x_ticks = ax.get_xticklabels()
+        translated_labels = [values_to_translations.get(label.get_text(), label.get_text()) for label in x_ticks]
+        ax.set_xticklabels(translated_labels)
+
+    ax.set_xticklabels(ax.get_xticklabels(), rotation=45, ha='right')
```

Comment:
I see, the translations are only applied here, during the creating of the plot.
Meaning that everywhere else it is internally using the value, which is good :)

Reply:
I was initially thrown off by
```python
values_to_translations.get(label.get_text(), label.get_text()
```
but I understand now.

`label.get_text()` is the "value" and we are searching `values_to_translations` for a mapping. But if that mapping doesn't exist (like with 'other' or a custom mode or something) then it defaults to the raw value